gateway/ipfs.go

```diff
@@ -79,23 +80,30 @@ func (g *Gateway) renderIPFSPath(c *gin.Context, base, pth string) {
 				"Links":   links,
 			}
 			c.HTML(http.StatusOK, "/public/html/unixfs.gohtml", params)
-		} else {
-			renderError(c, http.StatusBadRequest, err)
 			return
 		}
-	} else {
-		contentType := http.DetectContentType(data)
-		c.Writer.Header().Set("Content-Type", contentType)
-		c.Render(200, render.Data{Data: data})
+
+		renderError(c, http.StatusBadRequest, err)
+		return
+	}
+	defer f.Close()
+
+	var buf [512]byte
+	n, err := io.ReadAtLeast(f, buf[:], len(buf))
+	if err != nil && err != io.ErrUnexpectedEOF {
+		renderError(c, http.StatusInternalServerError, err)
+		return
 	}
+	contentType := http.DetectContentType(buf[:])
+	c.Writer.Header().Set("Content-Type", contentType)
+	c.Render(200, render.Reader{ContentLength: -1, Reader: io.MultiReader(bytes.NewReader(buf[:n]), f)})
```

Comment:
Since we're returning a reader, we can't pre-calculate the content-length of the response.

That's fine, but we should be explicit with content-length -1 since I looked up the `c.Render` impl, and that's the only way to signal `net/http` server that this return should be chunk encoded.